packages/yarnpkg-fslib/sources/url.ts

```diff
@@ -0,0 +1,90 @@
+import {URL, fileURLToPath as fileURLToPathImpl, pathToFileURL as pathToFileURLImpl} from 'url';
+
+import * as errors                                                                   from './errors';
+import {NativePath, PortablePath, Path, PathUtils, PathLike, PathType, ppath, npath} from './path';
+
+const PROTOCOL = `file:`;
+
+export class BaseFileURL<P extends Path> extends URL {
+  declare public protocol: typeof PROTOCOL;
```

Comment:
Same as the other file

Reply:
It (kinda?) makes sense to initialize this one to `PROTOCOL`, although it will get overridden by the `super` call. I'd personally keep it like this, since its only purpose is to augment the `protocol` property from the superclass. I think it would be better to also preserve the `declare` here, since it would lead to less unneeded runtime code. What do you think?